apis/core/v1beta1/featureflagsource_types.go

```diff
@@ -234,40 +214,45 @@ func (fc *FeatureFlagSourceSpec) ToEnvVars() []corev1.EnvVar {
 		})
 	}
 
-	if fc.ManagementPort != DefaultManagementPort {
+	if fc.ManagementPort != common.DefaultManagementPort {
 		envs = append(envs, corev1.EnvVar{
-			Name:  common.EnvVarKey(fc.EnvVarPrefix, SidecarMetricPortEnvVar),
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, common.ManagementPortEnvVar),
 			Value: fmt.Sprintf("%d", fc.ManagementPort),
 		})
 	}
 
-	if fc.Port != defaultPort {
+	if fc.Port != common.DefaultPort {
 		envs = append(envs, corev1.EnvVar{
-			Name:  common.EnvVarKey(fc.EnvVarPrefix, SidecarPortEnvVar),
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, common.PortEnvVar),
 			Value: fmt.Sprintf("%d", fc.Port),
 		})
 	}
 
-	if fc.Evaluator != defaultEvaluator {
+	if fc.Evaluator != common.DefaultEvaluator {
 		envs = append(envs, corev1.EnvVar{
-			Name:  common.EnvVarKey(fc.EnvVarPrefix, SidecarEvaluatorEnvVar),
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, common.EvaluatorEnvVar),
 			Value: fc.Evaluator,
 		})
 	}
 
-	if fc.SocketPath != defaultSocketPath {
+	if fc.SocketPath != "" {
 		envs = append(envs, corev1.EnvVar{
-			Name:  common.EnvVarKey(fc.EnvVarPrefix, SidecarSocketPathEnvVar),
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, common.SocketPathEnvVar),
 			Value: fc.SocketPath,
 		})
 	}
 
-	if fc.LogFormat != defaultLogFormat {
+	if fc.LogFormat != common.DefaultLogFormat {
 		envs = append(envs, corev1.EnvVar{
-			Name:  common.EnvVarKey(fc.EnvVarPrefix, SidecarLogFormatEnvVar),
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, common.LogFormatEnvVar),
 			Value: fc.LogFormat,
 		})
 	}
 
+	envs = append(envs, corev1.EnvVar{
```

Comment:
```suggestion
        // sets the FLAGD_RESOLVER var to "rpc" to configure the provider for RPC evaluation mode
	envs = append(envs, corev1.EnvVar{
```

Suggesting this comment because this is particularly important.